applications/minotari_console_wallet/src/ui/state/wallet_event_monitor.rs

```diff
@@ -221,7 +208,10 @@ impl WalletEventMonitor {
                     }
                 },
                 _ = base_node_changed.changed() => {
-                    let peer = base_node_changed.borrow().as_ref().cloned();
+                    let peer = {
+                        // Ensure the watch borrow is dropped immediately after use
+                        base_node_changed.borrow().as_ref().cloned()
+                    };
```

Comment:
Borrowed reference is dropped at this line, no need for an outer scope

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1b53c09b9b31090a303ec9580d1e1f39